edb/edgeql/compiler/context.py

```diff
@@ -537,6 +537,10 @@ class ContextLevel(compiler.ContextLevel):
     active_computeds: ordered.OrderedSet[s_pointers.Pointer]
     """A ordered set of currently compiling computeds"""
 
+    disallow_dml: bool
```

Comment:
Maybe make this an `Optional[str]` so we can say "UPDATE may not be done in a FILTER clause" or whatever